cmake/dependencies.cmake

```diff
@@ -0,0 +1,225 @@
+# find required system dependencies via pkg-config
```

Comment:
+1 for this, I could even recommend to put each external project config in a separate cmake file.

Reply:
I think that might be a bit overkill.

What I want to do though, is define some kind of pseudo targets, so that the checks in `src/` whether to use a system library to add a dependency on building the external projects. Also I hate having to work with stuff like `${GIO_LIBRARIES}`. It'd be much better if that would be abstracted entirely away in `dependencies.cmake`.